torch/csrc/autograd/variable.cpp

```diff
@@ -154,6 +154,25 @@ AutogradMeta* materialize_autograd_meta(const at::TensorBase& self) {
   return get_autograd_meta(self);
 }
 
+void update_retains_grad(
+    const at::TensorBase& self,
+    const std::shared_ptr<torch::autograd::Node>& fn) {
+  if (!self.retains_grad()) {
+    return;
+  }
+  auto idx = impl::get_autograd_meta(self)->retains_grad_;
+  auto old_list = materialize_autograd_meta(self)->cpp_hooks_list_;
```

Comment:
Not need to materialize here. You segfaulted on the line above if it wasn't there already.

Reply:
Good catch, somehow missed this lol